pkg/smokescreen/smokescreen.go

```diff
@@ -314,9 +318,20 @@ func dialContext(ctx context.Context, network, addr string) (net.Conn, error) {
 	}
 	sctx.logger = sctx.logger.WithFields(fields)
 
-	// Only wrap CONNECT conns with an InstrumentedConn. Connections used for traditional HTTP proxy
+	// Only wrap CONNECT conns and MITM http conns with an InstrumentedConn. Connections used for traditional HTTP proxy
 	// requests are pooled and reused by net.Transport.
-	if sctx.proxyType == connectProxy {
+	if sctx.proxyType == connectProxy || pctx.ConnectAction == goproxy.ConnectMitm {
+		// If we have a MITM and option is enabled, we can add detailed Request log fields
+		if pctx.ConnectAction == goproxy.ConnectMitm && sctx.Decision.MitmConfig != nil && sctx.Decision.MitmConfig.DetailedHttpLogs {
```

Comment:
Does the `MitmConfig` reference need to live with the decision? Could we not move this directly into the context?

Reply:
`Decision.MitmConfig` comes from [ACLDecision.MitmConfig](https://github.com/stripe/smokescreen/blob/f4ef674198764efd9e48128eaa365b865db1f177/pkg/smokescreen/smokescreen.go#L1027) because the config is in the rule which comes from `rule.DomainMitmGlobs`
Al these happen in the decision functions (`checkIfRequestShouldBeProxied`, `checkACLsForRequest`, `config.EgressACL.Decide` and none of these have any context.
Do you think we should pass `pctx` down to all these methods to set `MitmConfig` there?